File: bench/ruby/bintree.rb
n = (ENV["BENCH_N"] || "200").to_i   # repetitions
depth = 12

def make(d)
  return nil if d == 0
  [make(d - 1), make(d - 1)]
end

def check(node)
  return 1 if node.nil?
  1 + check(node[0]) + check(node[1])
end

total = 0
n.times { total += check(make(depth)) }

puts total

File: bench/go/bintree.go
package main

import "fmt"

type node struct{ l, r *node }

func mk(d int) *node {
	if d == 0 {
		return nil
	}
	return &node{mk(d - 1), mk(d - 1)}
}

func check(t *node) int {
	if t == nil {
		return 1
	}
	return 1 + check(t.l) + check(t.r)
}

func main() {
	n := benchN(200) // repetitions
	const depth = 12
	total := 0
	for i := 0; i < n; i++ {
		total += check(mk(depth))
	}
	fmt.Println(total)
}
